SConstruct

```diff
@@ -1358,7 +1358,7 @@ static void terminate_handler()
 			('DXX_MAX_AXES_PER_JOYSTICK', user_settings.max_axes_per_joystick),
 			('DXX_MAX_BUTTONS_PER_JOYSTICK', user_settings.max_buttons_per_joystick),
 			('DXX_MAX_HATS_PER_JOYSTICK', user_settings.max_hats_per_joystick),
-			('DXX_USE_SDL_REDBOOK_AUDIO', int(not sdl2)),
+			('DXX_USE_SDL_REDBOOK_AUDIO', int(not sdl2 or sys.platform in ['win32', 'msys'])), # there is a win32 rbaudio now
```

Comment:
This is not correct.  `sys.platform` describes the system on which the program is built, not the system on which it will be run.  When I cross-compile from Linux to produce a program for Windows users to run, this should default to true, but `sys.platform` in that case would be `linux2`.

Reply:
Yes, it should check the target environment, but I'm not sure how to achieve that.

Reply:
Make the default value a function, not an inline value.  When a default is needed, the function will be called.  That function can inspect settings that were defined above the setting for which this default is being computed.  Host platform is one of the first settings defined, so that functions for computing defaults can inspect it.  See `__default_DATA_DIR` for an example (though unfortunately not a very obvious example).

https://github.com/dxx-rebirth/dxx-rebirth/blob/d0566119891c7f381cbe7a375c09ccda6c0c689a/SConstruct#L3573-L3578

Reply:
The value is set in a completely different context for `DXX_USE_SDL_REDBOOK_AUDIO` than `sharepath`, and it seems the former would not have access to `self._program` or a similar structure.

Reply:
This makes it even easier, then.  The defines immediately above the changed line are based on `user_settings`, and `user_settings` is where you can find `host_platform`.  I think you could do this:

```diff
--- a/SConstruct
+++ b/SConstruct
@@ -1358,7 +1358,7 @@
 			('DXX_MAX_AXES_PER_JOYSTICK', user_settings.max_axes_per_joystick),
 			('DXX_MAX_BUTTONS_PER_JOYSTICK', user_settings.max_buttons_per_joystick),
 			('DXX_MAX_HATS_PER_JOYSTICK', user_settings.max_hats_per_joystick),
-			('DXX_USE_SDL_REDBOOK_AUDIO', int(not sdl2 or sys.platform in ['win32', 'msys'])), # there is a win32 rbaudio now
+			('DXX_USE_SDL_REDBOOK_AUDIO', int(not sdl2 or user_settings.host_platform == 'win32')), # there is a win32 rbaudio now
 		))
 		context.Display('%s: checking whether to enable joystick support...%s\n' % (self.msgprefix, 'yes' if user_settings.max_joysticks else 'no'))
 		# SDL2 removed CD-rom support.
```

I dropped the `msys` test since `host_platform` is `win32` even for msys.

https://github.com/dxx-rebirth/dxx-rebirth/blob/d0566119891c7f381cbe7a375c09ccda6c0c689a/SConstruct#L3652-L3659